.github/workflows/ci_tests.yaml

```diff
@@ -7,6 +7,7 @@ on:
   push:
     branches: [ master ]
   pull_request:
+    types: [opened, reopened, synchronize, ready_for_review]
```

Comment:
Could you explain these 'types' a bit more? I understand opened, [reopened](https://docs.github.com/en/developers/webhooks-and-events/issue-event-types#reopened) and [ready_for_review](https://docs.github.com/en/developers/webhooks-and-events/issue-event-types#ready_for_review), but what does synchronize do? I see 'synchronize is listed on https://docs.github.com/en/developers/webhooks-and-events/github-event-types#pullrequestevent but the docs don't say what that status means.

Reply:
`synchronize` is better explained in this issue https://github.com/github/docs/issues/2257.

FYI, the [documentation](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request) says, 

> By default, a workflow only runs when a pull_request's activity type is opened, synchronize, or reopened.

Here we need `ready_for_review` to trigger the full tests when pushing the button.

Reply:
Ah ok, sounds good then.